resources/metadata/transportRequestUploadRFC.yaml

```diff
@@ -48,60 +38,69 @@ spec:
       - name: username
         type: string
         mandatory: true
-        description: "Deploy user"
+        description: "Service user for uploading to the ABAP backend via RFC"
         secret: true
         scope:
           - PARAMETERS
           - STAGES
           - STEPS
           - GENERAL
+        resourceRef:
+          - name: uploadCredentialsId
+            type: secret
+            param: username
       - name: password
         type: string
         mandatory: true
-        description: "Password for the deploy user"
+        description: "Service user password for uploading to the ABAP backend via RFC"
         secret: true
         scope:
           - PARAMETERS
-      - name: applicationName
+        resourceRef:
+          - name: uploadCredentialsId
+            type: secret
+            param: password
+      - name: client
         type: string
-        mandatory: true
-        description: "Name of the application."
+        aliases:
+          - name: changeManagement/client
```

Comment:
Why did this line change from `changeManagement/rfc/developmentClient` to `changeManagement/client`? I have the same question for line 31 in the new version of the file (`changeManagement/rfc/developmentInstance` to `changeManagement/instance`)

Reply:
In the current implementation of the transports we have one step only for all types of endpoints: CTS/RFC/SOLMAN.
accordingly, general keys were introduced (default_pipeline_environment.yml) depending on the type, which is oart of the key path.

```
general:
  changeManagement:
    cts:
      osDeployUser: 'node'
      deployConfigFile: 'ui5-deploy.yaml'
    rfc:
      ...
    solman:
      ...
steps:
  transportRequestUploadFile:
      acceptUnixStyleLineEndings: true
      codePage: 'UTF-8'
      failOnWarning: true
      verbose: false
```

In the new/Go  implementation we have type specific steps, so the type isn't part of the key path of the `config.yml` anymore:

```
steps:
  transportRequestUploadRFC:
    changeManagement
      developmentClient: "001"
      ...
```

Unfortunately, i guess due to leagacy, the pathes aren't harmonized. e.g. changeManagement/client is for CTS only:

`            .withMandatoryProperty('changeManagement/client', null, {backendType == BackendType.CTS})
`
while changeManagement/rfc/developmentClient is for RFC only. Probably, these clients do not mean the same.

`            .withMandatoryProperty('changeManagement/rfc/developmentClient', null, { backendType == BackendType.RFC })
`

Well, by your question i started to doubt that I looked deep enough into the keys. I will check the old code and the doc again to make sure to support the old defaults since the transport steps like `transportRequestUploadRFC` inherit the configuration from `transportRequestUploadFile`.

```
metadata:
  name: transportRequestUploadRFC
  aliases:
    - name: transportRequestUploadFile
```

Reply:
I added the ordered aliases for compatibility. Should be fine now,